crates/driver/src/domain/competition/solution/settlement.rs

```diff
@@ -261,13 +261,16 @@ impl Settlement {
                 sell: self.solution.prices[&trade.order().sell.token.wrap(self.solution.weth)],
                 buy: self.solution.prices[&trade.order().buy.token.wrap(self.solution.weth)],
             };
-            order.sell = trade.sell_amount(&prices).unwrap_or_else(|err| {
+            order.side = trade.order().side;
```

Comment:
Overwriting the values returned by `.or_default()` seems pretty implicit.
We should enforce that we use the data from the original order if it was present in the auction or make use the additional data returned for JIT orders if it's not.
We should prevent the case where a solver can make up different values for an order that was already present in the auction.

Reply:
removed `.or_default`